### PR TITLE
Use first..last in ranges

### PR DESCRIPTION
**Pull request by eksperimental**
As opposed to `left..right` and `x..y`

### CODE REVIEW

**Comment by eksperimental** [comment]
This is the terminology used in the Range module
> Ranges can be either increasing (first <= last) or decreasing (first > last). Ranges are also always inclusive.

personally, I don't think it's confusing. if you have a range `100..1`, 100 is still the first integer in the range, regardless one being < than 100.

**Comment by yordis** [comment]
I would prefer to read `first` and `last` because @michalmuskala I see your issue as something that is relative what is `first` and `last` in your terminology (Relativity issue)

File: lib/elixir/lib/range.ex [hunk 1]
```diff
@@ -70,25 +70,20 @@ defimpl Enumerable, for: Range do
     reduce(first, last, acc, fun, _up? = last >= first)
   end
 
-  defp reduce(_x, _y, {:halt, acc}, _fun, _up?) do
-    {:halted, acc}
-  end
```
Comment (josevalim) [inline lib/elixir/lib/range.ex]: Please avoid changing the syntax. If the issue is about renaming variables, then it should keep the other syntax the same.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**